server/src/main/java/org/elasticsearch/search/vectors/KnnSearchBuilder.java

```diff
@@ -213,8 +216,8 @@ private KnnSearchBuilder(
 
     public KnnSearchBuilder(StreamInput in) throws IOException {
         this.field = in.readString();
-        this.k = in.readVInt();
-        this.numCands = in.readVInt();
+        this.k = in.readOptionalVInt();
+        this.numCands = in.readOptionalVInt();
```

Comment:
when you are ready for further review, etc. all these will have to be backwards compatible. Meaning looking at the `in.getTransportVersion()` and verifying its at least the version where this change is introduced.

Similar logic down in `writeTo(StreamOutput)` method.

Reply:
Thanks for the heads up @benwtrent ! Will update accordingly.